src/vs/workbench/electron-browser/window.ts

```diff
@@ -258,15 +258,21 @@ export class ElectronWindow {
 
 		// High Contrast Events
 		ipc.on('vscode:enterHighContrast', (event) => {
-			this.partService.joinCreation().then(() => {
-				this.themeService.setColorTheme(VS_HC_THEME, false);
-			});
+			const windowConfig = this.configurationService.getConfiguration<IWindowSettings>('window');
+			if (windowConfig.autoDetectHighContrast) {
+				this.partService.joinCreation().then(() => {
+					this.themeService.setColorTheme(VS_HC_THEME, false);
+				});
+			}
 		});
 
 		ipc.on('vscode:leaveHighContrast', (event) => {
-			this.partService.joinCreation().then(() => {
-				this.themeService.setColorTheme(VS_DARK_THEME, false);
-			});
+			const windowConfig = this.configurationService.getConfiguration<IWindowSettings>('window');
```

Comment:
I think we can ignore the setting in this case because leaving HC on Windows you want to switch back to dark theme. Or otherwise the setting should be clearer that it prevents switching in both ways, not just from normal to HC theme.